operators/vision/encode_image.cc

```diff
@@ -1,40 +1,124 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include "png.h"
+#include "jpeglib.h"
+#include "op_def_struct.h"
+#include "ext_status.h"
+
 #include "encode_image.hpp"
 
-#include <opencv2/imgcodecs.hpp>
 
 namespace ort_extensions {
 
 void KernelEncodeImage::Compute(const ortc::Tensor<uint8_t>& input, ortc::Tensor<uint8_t>& output) const {
-  // Setup inputs
-  const auto dimensions_bgr = input.Shape();
-
+  const auto& dimensions_bgr = input.Shape();
   if (dimensions_bgr.size() != 3 || dimensions_bgr[2] != 3) {
-    // expect {H, W, C} as that's the inverse of what decode_image produces.
-    // we have no way to check if it's BGR or RGB though
     ORTX_CXX_API_THROW("[EncodeImage] requires rank 3 BGR input in channels last format.", ORT_INVALID_ARGUMENT);
   }
 
-  // Get data & the length
   std::vector<int32_t> height_x_width{static_cast<int32_t>(dimensions_bgr[0]),   // H
                                       static_cast<int32_t>(dimensions_bgr[1])};  // W
+  const int color_space = 3;
+  const uint8_t* bgr_data = input.Data();
+  unsigned char* outbuffer = nullptr;
+  std::vector<uint8_t> png_buffer;
+  size_t outsize = 0;
+
+  auto rgb_data = std::make_unique<uint8_t[]>(height_x_width[0] * height_x_width[1] * color_space);
+  for (int y = 0; y < height_x_width[0]; ++y) {
+    for (int x = 0; x < height_x_width[1]; ++x) {
+      rgb_data[(y * height_x_width[1] + x) * color_space + 0] = bgr_data[(y * height_x_width[1] + x) * color_space + 2];
```

Comment:
perhaps we can save `height_x_width[1]` in a variable here since it is called many times, to save compute/time to access the vector each time.